pkg/util/helper/workstatus.go

```diff
@@ -84,22 +85,34 @@ func assembleWorkStatus(c client.Client, selector labels.Selector, workload *uns
 			klog.Errorf("Failed to get manifestIndex of workload in work.Spec.Workload.Manifests. Error: %v.", err)
 			return nil, err
 		}
+		clusterName, err := names.GetClusterName(work.Namespace)
+		if err != nil {
+			klog.Errorf("Failed to get clusterName from work namespace %s. Error: %v.", work.Namespace, err)
+			return nil, err
+		}
+
+		// if sync work to member cluster failed, then set status back to resource binding.
+		for _, c := range work.Status.Conditions {
+			if c.Status == v1.ConditionFalse {
+				aggregatedStatus := workv1alpha1.AggregatedStatusItem{
+					ClusterName:    clusterName,
+					Applied:        false,
+					AppliedMessage: c.Message,
+				}
+				statuses = append(statuses, aggregatedStatus)
+			}
+		}
 
 		for _, manifestStatus := range work.Status.ManifestStatuses {
 			equal, err := equalIdentifier(&manifestStatus.Identifier, identifierIndex, workload)
 			if err != nil {
 				return nil, err
 			}
 			if equal {
-				clusterName, err := names.GetClusterName(work.Namespace)
-				if err != nil {
-					klog.Errorf("Failed to get clusterName from work namespace %s. Error: %v.", work.Namespace, err)
-					return nil, err
-				}
-
 				aggregatedStatus := workv1alpha1.AggregatedStatusItem{
```

Comment:
ManifestStatuses will be empty for resource without status. For these resource, we should collect apply result

Reply:
Wow, is that true?